src/collector.rs

```diff
@@ -1,63 +1,282 @@
+        }
+    }
+
+    async fn url_contents(url: &Url) -> Result<InputContent> {
+        let res = reqwest::get(url.clone()).await?;
+        let content = res.text().await?;
+        let input_content = InputContent {
+            input: Input::RemoteUrl(url.clone()),
+            file_type: FileType::from(url.as_str()),
+            content,
+        };
+
+        Ok(input_content)
+    }
+
+    async fn glob_contents(path_glob: &str, ignore_case: bool) -> Result<Vec<InputContent>> {
+        let mut contents = vec![];
+        let glob_expanded = tilde(&path_glob);
+        let mut match_opts = glob::MatchOptions::new();
+
+        match_opts.case_sensitive = !ignore_case;
+
+        for entry in glob_with(&glob_expanded, match_opts)? {
+            match entry {
+                Ok(path) => {
+                    let content = Self::path_content(&path).await?;
+                    contents.push(content);
+                }
+                Err(e) => println!("{:?}", e),
+            }
+        }
+
+        Ok(contents)
+    }
+
+    async fn path_content<P: Into<PathBuf> + AsRef<Path>>(path: P) -> Result<InputContent> {
+        let input_content = InputContent {
+            file_type: FileType::from(path.as_ref()),
+            content: read_to_string(&path).await?,
+            input: Input::FsPath(path.into()),
+        };
+
+        Ok(input_content)
+    }
+
+    async fn stdin_content(file_type_hint: Option<FileType>) -> Result<InputContent> {
+        let mut content = String::new();
+        let mut stdin = stdin();
+        stdin.read_to_string(&mut content).await?;
+
+        let input_content = InputContent {
+            input: Input::Stdin,
+            file_type: file_type_hint.unwrap_or_default(),
+            content,
+        };
+
+        Ok(input_content)
+    }
+
+    fn string_content(s: &str, file_type_hint: Option<FileType>) -> Result<InputContent> {
+        Ok(InputContent::from_string(
+            s,
+            file_type_hint.unwrap_or_default(),
+        ))
     }
 }
 
-/// Fetch all unique links from a vector of inputs
+impl ToString for Input {
+    fn to_string(&self) -> String {
+        match self {
+            Self::RemoteUrl(url) => url.to_string(),
+            Self::FsGlob { pattern, .. } => pattern.clone(),
+            Self::FsPath(p) => p.to_str().unwrap_or_default().to_owned(),
+            Self::Stdin => STDIN.to_owned(),
+            Self::String(s) => s.clone(),
+        }
+    }
+}
+
+/// Fetch all unique links from a slice of inputs
 /// All relative URLs get prefixed with `base_url` if given.
 pub(crate) async fn collect_links(
-    inputs: Vec<String>,
+    inputs: &[Input],
     base_url: Option<String>,
+    skip_missing_inputs: bool,
 ) -> Result<HashSet<Uri>> {
     let base_url = match base_url {
         Some(url) => Some(Url::parse(&url)?),
         _ => None,
     };
 
-    let mut links = HashSet::new();
+    let (contents_tx, mut contents_rx) = tokio::sync::mpsc::channel(32);
```

Comment:
Guess we could use `MAX_CONCURRENCY` instead of 32 here to make that configurable? It's not a big deal, though.

Reply:
Sure, I was going to add it, but managed to overlook it :sweat: